test/extended/images/mysql_replica.go

```diff
@@ -0,0 +1,266 @@
+package images
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/util/wait"
+)
+
+func selector(selector string) labels.Selector {
+	ret, err := labels.Parse(selector)
+	if err != nil {
+		g.Fail(fmt.Sprintf("Couldn't parse selector %v: %v", selector, err))
+	}
+	return ret
+}
+
+// getPods looks up pods with "oc get" applying a label selector and returns
+// names of matching pods
+func getPods(oc *exutil.CLI, label labels.Selector) (pods []string, err error) {
+	podList, err := oc.KubeREST().Pods(oc.Namespace()).List(label, nil)
+	if err != nil {
+		return nil, err
+	}
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == kapi.PodRunning {
+			pods = append(pods, pod.Name)
+		}
+	}
+	return pods, nil
+}
+
+// pollForPods will call getPods in a loop until the required number of pods
+// is returned
+func pollForPods(oc *exutil.CLI, label labels.Selector, count int, timeout time.Duration) ([]string, error) {
+	var pods []string
+	err := wait.Poll(1*time.Second, timeout, func() (bool, error) {
+		p, e := getPods(oc, label)
+		if e != nil {
+			return true, e
+		}
+		if len(p) < count {
+			return false, nil
+		}
+		pods = p
+		return true, nil
+	})
+	return pods, err
+}
+
+type mysql struct {
+	oc      *exutil.CLI
+	PodName string
+	Env     map[string]string
+}
+
+func NewMysql(oc *exutil.CLI, podName, service string) (*mysql, error) {
+	mysql := &mysql{
+		oc:      oc,
+		PodName: podName,
+	}
+
+	pod, err := oc.KubeREST().Pods(oc.Namespace()).Get(podName)
+	if err != nil {
+		return nil, err
+	}
+	mysql.Env = make(map[string]string, len(pod.Spec.Containers[0].Env))
+	for _, env := range pod.Spec.Containers[0].Env {
+		mysql.Env[env.Name] = env.Value
+	}
+	return mysql, nil
+}
+
+// WaitForMySQL pings the MySQL server
+func (m *mysql) WaitForMySQL() (bool, error) {
+	out, err := m.oc.ExecWithoutExit(m.PodName, "mysqladmin -h "+"127.0.0.1"+" -u"+"root"+" ping").Output()
```

Comment:
Wait wait.. this should be a single string... no need for `fmt` nor concatenation:
```go
"mysqladmin -h 127.0.0.1 -uroot ping"
```